optuna/trial.py

```diff
@@ -110,9 +110,10 @@ class Trial(BaseTrial):
     This object is passed to an objective function and provides interfaces to get parameter
     suggestion, manage the trial's state, and set/get user-defined attributes of the trial.
 
-    Note that this object is seamlessly instantiated and passed to the objective function behind
-    :func:`optuna.study.Study.optimize()` method (as well as optimize function); hence, in typical
-    use cases, library users do not care about instantiation of this object.
+    Note that the direct use of this constructor is not recommended.
+    This object is seamlessly instantiated and passed to the objective function behind
+    :func:`optuna.study.Study.optimize()` method (as well as optimize function); hence
```

Comment:
Just out of curiosity, but what does the "optimize function" here refer to?

Reply:
To be honest, I don't know what it refers to...
I think that the sentence "(as well as optimize function)" can be removed. Please let me know if you have any thoughts about it. > @toshihikoyanase

Reply:
@sile @hvy
"optimize function" does not exist anymore. Thank you for pointing it out.

Reply:
Thanks! 
Removed: https://github.com/pfnet/optuna/pull/668/commits/4bac77d8c9d2523eda1208444cba198578656ded